instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanSuppressionStrategy.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import static java.util.Collections.singletonList;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import java.util.List;
+
+abstract class SpanSuppressionStrategy {
+  private static final SpanSuppressionStrategy SERVER_STRATEGY =
+      new SuppressIfSameSpanKey(singletonList(SpanKey.SERVER));
```

Comment:
Intellij is giving me some weird warnings here that I haven't seen before:

> Referencing subclass SuppressIfSameSpanKey from superclass SpanKeyStrategy initializer might lead to class loading deadlock

probably worth splitting the nested classes out to top-level classes